temporalio/workflow.py

```diff
@@ -5074,3 +5105,151 @@ def _to_proto(self) -> temporalio.bridge.proto.common.VersioningIntent.ValueType
+        input: InputT,
+        *,
+        output_type: Optional[Type[OutputT]] = None,
+        schedule_to_close_timeout: Optional[timedelta] = None,
+        headers: Optional[Mapping[str, str]] = None,
+    ) -> OutputT: ...
+
+
+class _NexusClient(NexusClient[ServiceT]):
+    def __init__(
+        self,
+        *,
+        endpoint: str,
+        service: Union[Type[ServiceT], str],
+    ) -> None:
+        """Create a Nexus client.
+
+        Args:
+            service: The Nexus service.
+            endpoint: The Nexus endpoint.
+        """
+        # If service is not a str, then it must be a service interface or implementation
+        # class.
+        if isinstance(service, str):
+            self.service_name = service
+        elif service_defn := nexusrpc.get_service_definition(service):
+            self.service_name = service_defn.name
+        else:
+            raise ValueError(
+                f"`service` may be a name (str), or a class decorated with either "
+                f"@nexusrpc.handler.service_handler or @nexusrpc.service. "
+                f"Invalid service type: {type(service)}"
+            )
+        self.endpoint = endpoint
+
+    # TODO(nexus-prerelease): overloads: no-input, ret type
+    # TODO(nexus-prerelease): should it be an error to use a reference to a method on a class other than that supplied?
+    async def start_operation(
+        self,
+        operation: Union[nexusrpc.Operation[InputT, OutputT], str, Callable[..., Any]],
+        input: InputT,
+        *,
+        output_type: Optional[Type[OutputT]] = None,
+        schedule_to_close_timeout: Optional[timedelta] = None,
+        headers: Optional[Mapping[str, str]] = None,
+    ) -> temporalio.workflow.NexusOperationHandle[OutputT]:
+        return (
+            await temporalio.workflow._Runtime.current().workflow_start_nexus_operation(
+                endpoint=self.endpoint,
+                service=self.service_name,
+                operation=operation,
+                input=input,
+                output_type=output_type,
+                schedule_to_close_timeout=schedule_to_close_timeout,
+                headers=headers,
+            )
+        )
+
+    # TODO(nexus-prerelease): overloads: no-input, ret type
+    async def execute_operation(
+        self,
+        operation: Union[nexusrpc.Operation[InputT, OutputT], str, Callable[..., Any]],
+        input: InputT,
+        *,
+        output_type: Optional[Type[OutputT]] = None,
+        schedule_to_close_timeout: Optional[timedelta] = None,
+        headers: Optional[Mapping[str, str]] = None,
+    ) -> OutputT:
+        handle = await self.start_operation(
+            operation,
+            input,
+            output_type=output_type,
+            schedule_to_close_timeout=schedule_to_close_timeout,
+            headers=headers,
+        )
+        return await handle
+
+
+def create_nexus_client(
+    endpoint: str, service: Union[Type[ServiceT], str]
```

Comment:
I would consider not making endpoint a positional so we can add a client that targets a task queue in the future.